lib/next_ls/helpers/ast_helpers.ex

```diff
@@ -204,20 +204,89 @@ defmodule NextLS.ASTHelpers do
     end
   end
 
-  def find_cursor(ast) do
-    with nil <-
-           ast
-           |> Zipper.zip()
-           |> Zipper.find(fn
-             {:@, _, [{:__cursor__, _, []}]} -> true
-             {:__cursor__, _, _} -> true
-             {{:., _, [_, :__cursor__]}, _, _} -> true
-             _ -> false
-           end) do
-      {:error, :not_found}
-    else
-      zipper -> {:ok, zipper}
-    end
+  def find_cursor(ast, position) do
+    ast = {:__block__, [], [ast]}
+
+    {tree, id} =
+      ast
+      |> Zipper.zip()
+      |> Zipper.traverse(0, fn tree, id ->
```

Comment:
probably want to keep an accumulator of the zipper when its at the spot it inserts the id, then instead of the second pass, we'll already be at the spot on the zipper to inject the cursor